cluster-creation-cli/README.md

```diff
@@ -0,0 +1,163 @@
+- The command should work the same way across all providers we support with Cluster API.
+- The command code should require little maintenance. It should stay compatible with a large range of cluster app versions.
+
+## Target use case
+
+The most important use case we target with our CLI command is the creation of a short-lived cluster for testing purposes, as this is the likely the most frequent reason to create a cluster.
+
+Production clusters, in contrast, are created much more rarely, often based on the experience gathered through testing with various clusters. Production clusters are also more likely to be provisioned through GitOps, while test clusters are often created and deleted ad-hoc.
+
+## Requirements
+
+- Users should be able to **re-use a configuration** they have used for cluster creation previously, optionally overriding only a few details.
+
+- A **dry run option** should allow "previewing" the result without actually writing any resources to the management cluster.
+
+- A **connection with the management cluster** should only be required to write resources. For a dry run, no connection should be necessary. (However, an internet connection will be required to access public information from GitHub repositories.)
+
+- Configuration provided by the user should be **validated** against the apps's schema, and in case of validation errors, users should receive detailed feedback.
+
+- Users should decide whether to actually create the cluster, or print a manifest only.
+
+- A **cluster name must be provided by the user**. We won't generate a cluster name for the user.
+  - At Giant Swarm we decided around November 2023 that we want users to set workload cluster names and avoid auto-generated names ([roadmap#2999](https://github.com/giantswarm/roadmap/issues/2999)). In theory, we could still offer an option (a flag) to auto-generate the name. However, this would require knowledge of the cluster-PROVIDER app schema in the command logic. For example, as of cluster-aws v0.73.0, the cluster name is set in the property .global.metadata.name. Although we are working towards alignment across providers, this may be different in other versions or other provider apps.
+
+## Basic syntax
+
+The command should be named `create cluster`, hence the entire command, without any flags, would be:
+
+    kubectl gs create cluster
+
+The most important flags would be:
+
+- `--provider`: (required) provider name. For example: capa, capvcd, capz, eks.
+  - TODO: How do we restrict and validate the provider list per installation?
+- `--config`: Path to a config file (optional). Can be used multiple times. See "Configuration layering" below. As an alternative, or in addition, configuration can be passed via standard input.
+- `--app-version`: the cluster-PROVIDER app version to use. If not specified, the latest version is used.
```

Comment:
we need to see how this would interact with the proposed release concept where the cluster version is defined by a release cc @nprokopic

Reply:
I think this is settled now. `--app-version` got replaced by `--release`.